.github/workflows/ci_allowance.yml

```
@@ -0,0 +1,43 @@
+name: safe-modules-allowance
+on:
+  push:
+    paths:
+      - 'modules/allowances/**'
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install pnpm
+        uses: pnpm/action-setup@v3
+        with:
+          version: 9
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+          cache: pnpm
+      - run: |
+          pnpm install
+          pnpm run --filter "@safe-global/safe-allowance-module" coverage
+      - uses: coverallsapp/github-action@master
+        with:
+          path-to-lcov: modules/allowances/coverage/lcov.info
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install pnpm
+        uses: pnpm/action-setup@v3
+        with:
+          version: 9
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+          cache: pnpm
+      - run: |
+          pnpm install
+          pnpm run --filter "@safe-global/safe-allowance-module" lint
+          pnpm run --filter "@safe-global/safe-allowance-module" fmt 
```

```suggestion
          pnpm run --filter "@safe-global/safe-allowance-module" fmt:check 
```

And in `package.json` we add a `"fmt:check": "prettier --check contracts/**/*.sol",`